ch_pipeline/core/telescope.py

```diff
@@ -203,13 +220,84 @@ def _load_layout(self):
 
         self._feeds = feeds
 
+    def _read_local_layout(self):
+        """Load the telescope layout from a file saved to this repository.
+
+        Note that this function will fallback to a database query if the
+        requested layout is earlier than September 1, 2018. As with
+        :method:`CHIME._load_layout`, this routine should not be called
+        directly.
+        """
+        # Do I/O, and resolve the layout, only on rank 0
+        if mpiutil.rank == 0:
+
+            # Get the path of the layout file
+            from importlib.resources import files
+
+            from . import telescope_files
+
+            layout_path = files(telescope_files).joinpath("layouts.pkl")
+
+            with layout_path.open("rb") as layout_f:
+                layouts = pickle.load(layout_f)
```

Comment:
```suggestion
            try:
                with layout_path.open("rb") as layout_f:
                    layouts = pickle.load(layout_f)
            except OSError as e:
                logger.warning(f"Failed to load local layout: {e}.  Will try to load layout from the database")
                return None  # Or whatever
```

If you do implement the fallback, I also think it would be a good idea to wrap the actual I/O here in a `try ... except OSError` block to catch I/O errors, which you could then handle by returning the failure result, meaning the pipeline will seamlessly fall back on trying to load the layout from the database in this case.

Reply:
This would also catch the case where the user has failed to properly install the `layouts.pkl` file.